File: app/src/main/java/com/example/myapplication/LoginSignup/PhoneNumber.java
package com.example.myapplication.LoginSignup;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageButton;

import com.example.myapplication.R;
import com.google.firebase.FirebaseApp;
import com.hbb20.CountryCodePicker;


public class PhoneNumber extends AppCompatActivity {

    private String fullNumber;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_phone_number);
        getSupportActionBar().hide();

        FirebaseApp.initializeApp(this);

        Button button = findViewById(R.id.goto_phone_verification);
        button.setOnClickListener(v -> openNext());

        ImageButton back = findViewById(R.id.backButton);
        back.setOnClickListener(v -> goBack());

        CountryCodePicker ccp = findViewById(R.id.ccp);
        EditText phoneNumber = findViewById(R.id.phoneNumber);
        ccp.registerCarrierNumberEditText(phoneNumber);

        ccp.setPhoneNumberValidityChangeListener(new CountryCodePicker.PhoneNumberValidityChangeListener() {
            @Override
            public void onValidityChanged(boolean isValidNumber) {
                if(isValidNumber) {
                    fullNumber = ccp.getFullNumberWithPlus();
                }
                button.setEnabled(ccp.isValidFullNumber());
            }
        });

    }

    public void openNext() {
        Intent intent = new Intent(getApplicationContext(), PhoneVerification.class);
        intent.putExtra("phoneNo", fullNumber);
        startActivity(intent);
    }

    public void goBack() {
        Intent intent = new Intent(this, StarUpScreen.class);
        startActivity(intent);
    }

}

File: app/src/main/java/com/example/myapplication/loginSignup/PhoneNumber.kt
package com.example.myapplication.loginSignup

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.ImageButton
import androidx.appcompat.app.AppCompatActivity
import com.example.myapplication.R
import com.google.firebase.FirebaseApp
import com.hbb20.CountryCodePicker

class PhoneNumber : AppCompatActivity() {
    private var fullNumber: String? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_phone_number)
        supportActionBar!!.hide()
        FirebaseApp.initializeApp(this)
        val button = findViewById<Button>(R.id.goto_phone_verification)
        button.setOnClickListener { openNext() }
        val back = findViewById<ImageButton>(R.id.backButton)
        back.setOnClickListener { goBack() }
        val ccp = findViewById<CountryCodePicker>(R.id.ccp)
        val phoneNumber = findViewById<EditText>(R.id.phoneNumber)
        ccp.registerCarrierNumberEditText(phoneNumber)
        ccp.setPhoneNumberValidityChangeListener { isValidNumber ->
            if (isValidNumber) {
                fullNumber = ccp.fullNumberWithPlus
            }
            button.isEnabled = ccp.isValidFullNumber
        }
    }

    private fun openNext() {
        val intent = Intent(applicationContext, PhoneVerification::class.java)
        intent.putExtra("phoneNo", fullNumber)
        startActivity(intent)
    }

    private fun goBack() {
        val intent = Intent(this, StartUpScreen::class.java)
        startActivity(intent)
    }
}